File: objective-c/The Logic/PauseViewController.m
//
//  PauseViewController.m
//  The Logic
//
//  Created by babi4_97 on 05.11.2017.
//  Copyright © 2017 Ivan Babkin. All rights reserved.
//

#import "PauseViewController.h"

@interface PauseViewController ()

@end

@implementation PauseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)resumeGame:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)playAgain:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
    [self.delegate startNewGame];
}

- (IBAction)mainMenu:(id)sender {
    [self.navigationController popToRootViewControllerAnimated:YES];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: The logic/PauseViewController.swift
//
//  PauseViewController.swift
//  The logic
//
//  Created by Ivan Babkin on 13.06.2018.
//  Copyright © 2018 Ivan Babkin. All rights reserved.
//

import UIKit

protocol PauseViewControllerDelegate: class {
    func startGame()
}

class PauseViewController: UIViewController {
    
    weak var delegate: PauseViewControllerDelegate!

    @IBAction func resume(_ sender: Any) {
        navigationController?.popViewController(animated: true)
    }
    
    @IBAction func playAgain(_ sender: Any) {
        navigationController?.popViewController(animated: true)
        delegate.startGame()
    }
    
    @IBAction func mainMenu(_ sender: Any) {
        navigationController?.popToRootViewController(animated: true)
    }
    
}
